jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,738 @@
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void include(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.include(new InetAddressPatternPredicate(p));
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void includeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            include(InetAddressPattern.from(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void exclude(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.exclude(new InetAddressPatternPredicate(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern} in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void excludeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            exclude(InetAddressPattern.from(p));
+    }
+
+    /**
+     * {@link IncludeExclude#include(Object) Include} arbitrary {@link Predicate}s in the conditions.
+     * @param predicates {@link Predicate}s that are tested against the {@link Request}.
+     * This method is optimized so that a passed {@link MethodPredicate} or {@link PathSpecPredicate} is
+     * converted to a more efficient {@link #includeMethod(String...)} or {@link #include(PathSpec...)} respectively.
+     */
+    @SafeVarargs
+    public final void include(Predicate<Request>... predicates)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (Predicate<Request> p : predicates)
+        {
+            if (p instanceof MethodPredicate methodPredicate)
+                includeMethod(methodPredicate._method);
+            else if (p instanceof PathSpecPredicate pathSpecPredicate)
+                include(pathSpecPredicate._pathSpec);
+            else
+                _predicates.include(p);
```

Comment:
Missing a branch for `InetAddressPatternPredicate`?

Reply:
No optimization available for inet addresses (yet?) just just put them in the linear predicate set